### PR TITLE
Markets page

### LINKED ISSUE #175: Markets page

### PR DESCRIPTION
**Pull request by nguvictor**
Fixes #175

### CODE REVIEW

File: src/app/components/markets/MarketsInfo.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+
+export function MarketsInfo() {
+  const marketPairs = 0;
+  const xrdVolume = 0;
+  const volumeChange = 0;
```
Comment (EvgeniiaVak) [inline src/app/components/markets/MarketsInfo.tsx:6]: these 0s do not look right, shouldn't we update them from somewhere?
Screenshot 2023-11-15 at 19 30 37

**Review by EvgeniiaVak — CHANGES_REQUESTED** [review]
* there should be link to `/markets` in the footer also
* trade now button doesn't work
* maybe making the lines between rows less prominent (less color contrast)? they kind of distract from the content, I would remove the one before the table header at all

If possible (I've been quite unable to debug much myself lately due to the wallet not connecting) please prioritize bug fixes over polishing this PR or adding other features.

File: src/app/components/markets/MarketsInfo.tsx (hunk 1 above)
Comment (nguvictor) [reply src/app/components/markets/MarketsInfo.tsx:6]: @fliebenberg would the alphadex api have these numbers somewhere or would we count/add this up on the client side?

Comment (fliebenberg) [reply src/app/components/markets/MarketsInfo.tsx:6]: Not at the moment, but I will add an API point to provide this kind of info. Am working on the API at the moment, so will add it and let you know once done. Hopefully today still.

Comment (fliebenberg) [reply src/app/components/markets/MarketsInfo.tsx:6]: I have added a new endpoint to the AlphaDEX api at /alphadex/market_info. It returns an object with the following fields:
- noPairs:  the number of pairs on the exchange
- noTokens: the number of tokens on the exchange
- xrdValueOfLiquidity: the XRD value of the total liquidity provided on the exchange
- noTrades24h: the number of completed trades on the exchange in the last 24h
- xrdValueOfTrades24h: the XRD value of completed trades in the last 24h

Let me know if this works for you.

File: src/app/components/markets/MarketsInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  return (
+    <div className="flex justify-between items-start">
```
Comment (EvgeniiaVak) [inline src/app/components/markets/MarketsInfo.tsx:9]:
```suggestion
    <div className="flexg-grow flex justify-between items-start">
```

The footer jumps up to the middle of the page. Can be fixed by adding `flex-grow` class to this div.

Screenshot 2024-01-26 at 18 16 34

File: src/app/components/markets/MarketsTable.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import { PairInfo } from "alphadex-sdk-js/lib/models/pair-info";
+
+const headers = [
+  { id: "id", label: "#" },
+  { id: "name", label: "TOKENS" },
+  { id: "lastPrice", label: "PRICE" },
+  { id: "quantity24h", label: "VOLUME 24H" },
+  { id: "change24h", label: "24H" },
+  //{ id: "high24h", label: "HIGH 24H" },
+  //{ id: "low24h", label: "LOW 24H" },
```
Comment (EvgeniiaVak) [inline src/app/components/markets/MarketsTable.tsx:15]: Do we need these comments?

File: src/app/components/markets/MarketsTable.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  const noDigits = 4;
+  const fixedDecimals = 3;
+  return pairs.map((pair) => {
+    return {
+      name: pair.name,
+      lastPrice: displayNumber(pair.lastPrice, noDigits, fixedDecimals),
+      quantity24h: displayNumber(pair.quantity24h, noDigits, fixedDecimals),
+      change24h: displayNumber(pair.change24h, noDigits, fixedDecimals),
+      //high24h: displayNumber (pair.high24h, noDigits, fixedDecimals),
+      //low24h: displayNumber (pair.low24h, noDigits, fixedDecimals),
```
Comment (EvgeniiaVak) [inline src/app/components/markets/MarketsTable.tsx:51]: and these?

File: src/app/markets/page.tsx [hunk 5]
```diff
@@ -1,3 +1,37 @@
...
+  return (
+    <div className="col-span-full p-4 mx-10">
+      <div className="flex mx-5 p-8 justify-between items-center uppercase border-b-2 border-b-secondary-content">
+        <div className="flex-none">
+          <MarketsInfo />
+        </div>
+        <div className="flex-last">
+          <button className="btn btn-lg uppercase bg-neutral text-accent border-x-4 border-y-0 border-accent">
+            Trade Now
+          </button>
```
Comment (EvgeniiaVak) [inline src/app/markets/page.tsx:29]: This button does not do anything. Maybe removing it to not create confusion? We already have the TRADE button in the navbar.

**Review by EvgeniiaVak — CHANGES_REQUESTED** [review]

**Comment by nguvictor** [comment]
Apologizes for the any extra pings here, just trying to bring this branch up to date.

**Review by EvgeniiaVak — CHANGES_REQUESTED** [review]

**Review by EvgeniiaVak — CHANGES_REQUESTED** [review]
I got a review notification from this PR, is this ready for review (not a draft anymore)?

There are eslint errors about unused imports:
```
/home/runner/work/website/website/src/app/components/markets/MarketsInfo.tsx
Error:   3:10  error    'displayNumber' is defined but never used        @typescript-eslint/no-unused-vars
Error:   [6](https://github.com/DeXter-on-Radix/website/actions/runs/7991497786/job/21822786836?pr=194#step:6:7):9   error    'marketInfo' is assigned a value but never used  @typescript-eslint/no-unused-vars
Error:   6:3[8](https://github.com/DeXter-on-Radix/website/actions/runs/7991497786/job/21822786836?pr=194#step:6:9)  error    'state' is defined but never used                @typescript-eslint/no-unused-vars
```

**Comment by nguvictor** [comment]
Hey, I was updating it. I'm planning to finish this in the next day or two, just final touches. Still a draft atm

**Comment by dcts** [comment]
Looks like this is outdated, closing to cleanup. Feel free to reopen if needed.

**Pull request closed without merging**